### PR TITLE
WIP: allow change day/resoruce grouping logic

### PR DESCRIPTION
**Pull request by jiangzhu-genie**
This PR is for the feature request https://github.com/intljusticemission/react-big-calendar/issues/1026

initial implementation for communication purpose

all comments, suggestions are welcome

storybook - resource group by day first

### CODE REVIEW

File: .storybook/webpack.config.js [hunk 1]
```diff
@@ -14,6 +14,6 @@ module.exports = function(config) {
         rules.less({ browsers }),
       ],
     },
-    plugins: [...config.plugins, plugins.extractText()],
+    // plugins: [...config.plugins, plugins.extractText()],
```
Comment (jiangzhu-genie) [inline .storybook/webpack.config.js]: Storybook cannot start on my local, it complains plugins.extractText is not a function, commented out temporarily

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (jiangzhu-genie) [inline .vscode/settings.json]: will remove when out of WIP

File: src/TimeGridHeader.js [hunk 3]
```diff
@@ -128,21 +119,58 @@ class TimeGridHeader extends React.Component {
...
+          {this.renderHeaderCell({ date, index, today })}
+        </div>
+        <div className="rbc-row rbc-row-resource">
+          {resources.map(([id, resource], idx) => (
+            <div key={`resource_${id || idx}`} className="rbc-header">
+              {accessors.resourceTitle(resource)}
+            </div>
+          ))}
+        </div>
+        {/* TODO - render date content row */}
```
Comment (jiangzhu-genie) [inline src/TimeGridHeader.js]: I need some advice on how to refactor existing code to allow render DateContentRow to support day/resource grouping

File: src/Calendar.js [hunk 4]
```diff
@@ -219,6 +219,14 @@ class Calendar extends React.Component {
      */
     resourceTitleAccessor: accessor,
 
+    /**
+     * Determines the group header of week view or work_week view with resources
+     *
+     * date - display date as group header and show resources as sub-headers
+     * resource - display resource as group header and show dates as sub-headers
+     */
+    resourceWeekViewHeader: PropTypes.oneOf(['day', 'resource']),
```
Comment (jiangzhu-genie) [inline src/Calendar.js]: Probably not the best named prop 😄
If have better name, please advice

File: stories/Resources.js [hunk 5]
```diff
@@ -4,6 +4,14 @@ import { storiesOf } from '@storybook/react'
 import { DragableCalendar } from './helpers'
 import resources from './helpers/resourceEvents'
 
-storiesOf('Resources').add('demo', () => (
-  <DragableCalendar events={resources.events} resources={resources.list} />
-))
+storiesOf('Resources')
```
Comment (jiangzhu-genie) [inline stories/Resources.js]: The repository seems doesn't have component tests e.g. enzyme + snapshot.

What should I do to test my changes on components?

**Comment by jquense** [comment]
I think you stumbled on the main reason why the current approach was taken. There is no easy way to refactoring DateContentRow to handle something like discrete row like the new grouping. 

Honestly, if we think about resizing or drag and drop as well the ux is unclear for the proposed grouping as well. I don't think it's probably worth the change in terms of added complexity :/

**Comment by jiangzhu-genie** [comment]
@jquense Thanks

I agree with your opinion from a developer point of view. In our use cases, there's no events across multiple days therefore for most of our users, it make sense to group by day first then resource.

I'll see what I can do with the DateContentRow in my own folk.

**Comment by TeaBough** [comment]
I'm really interested by this feature, did you managed to do something on your own @jiangzhu-genie ?

**Comment by jiangzhu-genie** [comment]
@TeaBough  not really, we did some investigation with our customers and most of them are happy with  Resource then Date grouping so this issue has been de-prioritised for us.

**Pull request closed without merging**